StellarDotnetSdk.Tests/Assets/AssetTest.cs

```diff
@@ -105,4 +134,302 @@ public void TestAssetEquals()
+        };
+        Asset.FromXdr(xdrAsset);
+    }
+
+    /// <summary>
+    /// Tests that native assets compare as equal (CompareTo returns 0).
+    /// Verifies that all native assets are considered equivalent for ordering purposes.
+    /// </summary>
+    [TestMethod]
+    public void TestCompareTo_NativeVsNative()
+    {
+        var native1 = new AssetTypeNative();
+        var native2 = new AssetTypeNative();
+        Assert.AreEqual(0, native1.CompareTo(native2));
+    }
+
+    /// <summary>
+    /// Tests comparison ordering between native assets and CreditAlphaNum4 assets.
+    /// Verifies that native assets sort before credit assets (native returns -1, credit returns 1).
+    /// </summary>
+    [TestMethod]
+    public void TestCompareTo_NativeVsCreditAlphaNum4()
+    {
+        var native = new AssetTypeNative();
+        var credit = new AssetTypeCreditAlphaNum4("USD", KeyPair.Random().AccountId);
+        Assert.AreEqual(-1, native.CompareTo(credit));
+        Assert.AreEqual(1, credit.CompareTo(native));
+    }
+
+    /// <summary>
+    /// Tests comparison ordering between native assets and CreditAlphaNum12 assets.
+    /// Verifies that native assets sort before credit assets regardless of code length.
+    /// </summary>
+    [TestMethod]
+    public void TestCompareTo_NativeVsCreditAlphaNum12()
+    {
+        var native = new AssetTypeNative();
+        var credit = new AssetTypeCreditAlphaNum12("TESTTEST", KeyPair.Random().AccountId);
+        Assert.AreEqual(-1, native.CompareTo(credit));
+        Assert.AreEqual(1, credit.CompareTo(native));
+    }
+
+    /// <summary>
+    /// Tests comparison ordering between CreditAlphaNum4 and CreditAlphaNum12 assets.
+    /// Verifies that CreditAlphaNum4 assets sort before CreditAlphaNum12 assets (4-character codes before 5-12 character codes).
+    /// </summary>
+    [TestMethod]
+    public void TestCompareTo_CreditAlphaNum4VsCreditAlphaNum12()
+    {
+        var issuer = KeyPair.Random().AccountId;
+        var credit4 = new AssetTypeCreditAlphaNum4("USD", issuer);
+        var credit12 = new AssetTypeCreditAlphaNum12("TESTTEST", issuer);
+        Assert.AreEqual(-1, credit4.CompareTo(credit12));
+        Assert.AreEqual(1, credit12.CompareTo(credit4));
+    }
+
+    /// <summary>
+    /// Tests that assets with the same code but different issuers are not equal.
+    /// Verifies that issuer is part of the comparison logic (same code from different issuers are different assets).
+    /// </summary>
+    [TestMethod]
+    public void TestCompareTo_CreditAlphaNum4_SameCodeDifferentIssuer()
+    {
+        var issuer1 = KeyPair.Random().AccountId;
+        var issuer2 = KeyPair.Random().AccountId;
+        var credit1 = new AssetTypeCreditAlphaNum4("USD", issuer1);
+        var credit2 = new AssetTypeCreditAlphaNum4("USD", issuer2);
+        var comparison = credit1.CompareTo(credit2);
+        Assert.IsTrue(comparison != 0);
```

Comment:
We can directly compare two assets, right?
```suggestion
        Assert.AreNotEqual(credit1, credit2);
```